src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/CategoryFilter.php

```diff
@@ -0,0 +1,155 @@
+{
+    /** @var CategoryRepositoryInterface */
+    protected $categoryRepository;
+
+    /**
+     * @param CategoryRepositoryInterface $categoryRepository
+     * @param array                       $supportedFields
+     * @param array                       $supportedOperators
+     */
+    public function __construct(
+        CategoryRepositoryInterface $categoryRepository,
+        array $supportedFields = [],
+        array $supportedOperators = []
+    ) {
+        $this->supportedFields = $supportedFields;
+        $this->supportedOperators = $supportedOperators;
+        $this->categoryRepository = $categoryRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addFieldFilter($field, $operator, $value, $locale = null, $scope = null, $options = [])
+    {
+        if ($operator !== Operators::UNCLASSIFIED) {
+            $this->checkValue($field, $value);
+        }
+
+        switch ($operator) {
+            case Operators::IN_LIST:
+                $clause = [
+                    'terms' => [
+                        'categories' => $value
+                    ]
+                ];
+
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::NOT_IN_LIST:
+                $clause = [
+                    'terms' => [
+                        'categories' => $value
+                    ]
+                ];
+
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+
+            case Operators::IN_CHILDREN_LIST:
+                $childrenCategoryCodes = $this->getAllChildrenCodes($value);
+                $clause = [
+                    'terms' => [
+                        'categories' => $childrenCategoryCodes
+                    ]
+                ];
+
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::NOT_IN_CHILDREN_LIST:
+                $childrenCategoryCodes = $this->getAllChildrenCodes($value);
+                $clause = [
+                    'terms' => [
+                        'categories' => $childrenCategoryCodes
+                    ]
+                ];
+
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+
+            case Operators::UNCLASSIFIED:
+                $clause = [
+                    'exists' => ['field' => 'categories']
+                ];
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+
+            case Operators::IN_LIST_OR_UNCLASSIFIED:
+                $shouldClauseInList = [
+                    'terms' => [
+                        'categories' => $value
+                    ],
+                ];
+                $shouldClauseUnclassified = [
+                    'bool' => [
+                        'must_not' => [
+                            'exists' => ['field' => 'categories']
```

Comment:
If I undestand well, ES will search all products where "categories" field does not exist. But what' happen if "categories" exists but is empty ? It will work ?